crates/sparrow-runtime/src/prepare/preparer.rs

```diff
@@ -40,14 +44,17 @@ impl Preparer {
         key_column_name: String,
         prepared_schema: SchemaRef,
         prepare_hash: u64,
-    ) -> Self {
-        Self {
+        time_unit: Option<&str>,
```

Comment:
Rust code changed to:

a) Accept a `time_unit` (to use for int64 and float64 times). It's optional, and currently defaults to `ns`. We could make it required (when time is a number) and/or change the default if we wish.
b) Instead of just casting the time column, it applies some logic to perform the multiplication.